src/com/amazon/ion/impl/IonReaderBinaryIncremental.java

```diff
@@ -1204,9 +1199,7 @@ public void stepIn() {
         }
         // Note: the IonReader interface dictates that stepping into a null container has the same behavior as
         // an empty container.
-        ContainerInfo containerInfo = containerStack.push();
-        containerInfo.type = valueType;
-        containerInfo.endPosition = valueEndPosition;
+        containerStack.push(c -> c.initialize(valueType, valueEndPosition));
```

Comment:
This is a change that would warrant performance testing of binary incremental reading to ensure it does not introduce a regression. Do we see extra overhead introduced by creation and invocation of a lambda on each stepIn?

Reply:
I will check if the change introduce regression on binary incremental reading.

Reply:
Here is the benchmark results of benchmarking incremental reader with 4 test data, and the results are neutral.  

Test Data | Before | After
-- | -- | --
log_59155.ion | 242.721 | 249.478
log_194617.ion | 1858.668 | 1854.447
test.json | 4273.663 | 4266.974
catalog.ion | 0.476 | 0.478
singleValue.10n | 0.063 | 0.063